### ISSUE TITLE
linter: False positive with react/exhaustive-deps when using a value as a type inside a generic

### ISSUE THREAD

**Issue by fknop** [oxc-project/oxc#11809]
### What version of Oxlint are you using?

1.2.0

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "plugins": [
    "typescript",
    "react",
    "oxc",
    "import",
    "node",
    "promise",
    "jest"
  ],
  "rules": {
    "react/exhaustive-deps": "error"
  }
}

```

### What happened?

Hello, 

We just started migrating to oxlint and we've seen some false positive with the react/exhaustive-deps rule.
When a value is used as a type in a generic inside the useMemo, it reports a false positive. However if it's used as a type declaration for a variable, it works fine, I've attached two screenshots of a minimal example.

This doesn't work:
![Image](https://github.com/user-attachments/assets/b2182781-3ed5-45d3-85ca-85ede4834007)

This works:
![Image](https://github.com/user-attachments/assets/45686b3e-2549-4d36-b1f6-8d8507a71034)
